pkg/statuscodes/statuscodes.go

```diff
@@ -68,6 +68,9 @@ const (
 	// Deprecated: In reality server-side errors should fall into one of the above 3 errors, and this inclusion was
 	// a mistake.  It's not worth a breaking change to revoke at this time, though, so it shall live on.
 	UnknownError StatusCode = 803
+	// DeadlineExceeded is for when the server is unable to complete the request within the configured deadline and the
+	// request times out. This error is retriable, but the duration for backoff is unknown.
+	DeadlineExceeded StatusCode = 804
```

Comment:
I guess returning timeouts as `Unavailable` can lead to some misunderstandings in the future as `Unavailable` is generally meant for a temporary downtime of the whole service and `DeadlineExceeded` is meant as a failure of just the one particular request. The [gRPC statuscodes](https://grpc.github.io/grpc/core/md_doc_statuscodes.html) also define it as two separate cases.

Reply:
I think even the FE can show two different messages to the user to better explain the situation.

* `Unavailable` - Service is temporarily unavailable, please try again later
* `DeadlineExceeded` - The request took too long, please try again later

In the second case the user can understand that the action he's performing is probably resource-intensive and can try to optimize the request, for example.

Reply:
That's a fair point. I'm for this 😄

Reply:
In my opinion the case does not need a dedicated status code.
If server failed to process user request within certain timeout it is an InternalServerError - transient condition that can and should be retried by the client

Eng reaction to it should either be 
- increase timeout because it is too low
- improve performance of the request processing (DB optimization, caching, etc...)

But this opinion is not blocking.

Reply:
My intention for this status code was mainly to track it separately in our gRPC dashboards but since it has wider implications I will start discussion with FE team and PM as well.